xrbk_macro/src/definition/expansion/derives.rs

```diff
@@ -0,0 +1,159 @@
+
+                // Expand the tokens to read each element.
+                let eqs = TokenStream2::with_tokens(|tokens| {
+                    for element in &self.content {
+                        element.partial_eq_tokens(tokens);
+                    }
+                });
+
+                tokens.append_tokens(quote_spanned!(trait_path.span()=>
+                    #[automatically_derived]
+                    impl #impl_generics ::std::cmp::PartialEq for #ident #type_generics #where_clause {
+				        #[allow(clippy::nonminimal_bool)]
+                        fn eq(&self, other: &Self) -> bool {
+                            // Default value which will be compared to all checked fields
+                            true
+
+                            // All the fields checks
+                            #eqs
+                        }
+                    }
+                ));
+            }
+
+            pub fn impl_hash(&self, tokens: &mut TokenStream2, trait_path: &Path) {
+                let ident = &self.ident;
+
+                // TODO: add generic bounds
+                let (impl_generics, type_generics, where_clause) = self.generics.split_for_impl();
+
+                // Expand the tokens to read each element.
+                let hashes = TokenStream2::with_tokens(|tokens| {
+                    for element in &self.content {
+                        element.hash_tokens(tokens);
+                    }
+                });
+
+                tokens.append_tokens(quote_spanned!(trait_path.span()=>
+                    #[automatically_derived]
+                    impl #impl_generics ::core::hash::Hash for #ident #type_generics #where_clause {
+                        fn hash<__H: ::core::hash::Hasher>(&self, state: &mut __H) {
+                            // All the hashes for all fiels
+                            #hashes
+                        }
+                    }
+                ));
+            }
+        }
+    };
+}
+
+structlike_impl_partial_eq!(Struct);
+structlike_impl_partial_eq!(Request);
+structlike_impl_partial_eq!(Reply);
+structlike_impl_partial_eq!(Event);
+
+impl Enum {
+	pub fn impl_partial_eq(&self, tokens: &mut TokenStream2, trait_path: &Path) {
+		let ident = &self.ident;
+
+		// TODO: add generic bounds
+		let (impl_generics, type_generics, where_clause) = self.generics.split_for_impl();
+
+		let arms = TokenStream2::with_tokens(|tokens| {
+			for variant in &self.variants {
+				let ident = &variant.ident;
+
+				let pat = TokenStream2::with_tokens(|tokens| {
+					variant.content.pat_cons_to_tokens(tokens);
+				});
+
+				let eqs = TokenStream2::with_tokens(|tokens| {
+					for element in &variant.content {
+						element.partial_eq_tokens(tokens);
+					}
+				});
+
+				tokens.append_tokens(quote_spanned!(trait_path.span()=>
+					Self::#ident #pat => {
+						// Default value which will be compared to all checked fields
+						true
```

Comment:
Yeah this comment is misleading.
It's just because `#eqs` contains all the field comparisons, which all starts with `&&`, so i need to put `true` here for it to work:
```rust
true
// All generated field comparisons
&& self.field1 == other.field1
&& self.field2 == other.field2
```

Reply:
I see.  I think in that case you would compare the enums something like:
```rust
match (&self, &other) {
    (Self::#ident #pat1, Self::#ident #pat2) => {
        // have a prefix for the fields in one of the patterns
        // so that you can bind them to different names.
    }
}
```
so that you can compare equality for enums in that:
(a) the variant has to match; and
(b) I don't think what you currently have would work to get the fields of the other enum, because you haven't matched against the enum variant.